### PR TITLE
Add discovery package

### PR DESCRIPTION
**Pull request by welcoMattic**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
         "php-http/message": "^1.3",
         "php-http/client-common": "^1.2",
         "php-http/cache-plugin": "^1.0",
-        "php-http/logger-plugin": "^1.0"
+        "php-http/logger-plugin": "^1.0",
+	"php-http/discovery": "^1.0"
```
Comment (sagikazarmark) [inline composer.json]: Please fix the indentation

**Comment by Nyholm** [comment]
Thank you for this PR. The discovery is needed. 
Could you check the indentation?

File: composer.json (hunk 1 above)
Comment (welcoMattic) [reply composer.json]: my bad, it's done ;)

**Comment by welcoMattic** [comment]
@Nyholm I thin client-common is needed too?

**Comment by Nyholm** [comment]
Client common is already required, right?

**Comment by welcoMattic** [comment]
In fact yes, but when I try to use `Httplug::sendRequest($request);`, I catch an error like:

```
Class 'Http\Adapter\Guzzle6\Client' not found
```

EDIT: I just run  `composer require php-http/guzzle6-adapter` and magic happens ;)

**Comment by Nyholm** [comment]
Yes, `composer require php-http/guzzle6-adapter` is what was missing in **your** app. 

Ill make sure to merge this PR as soon as the tests are green

**Comment by joelwurtz** [comment]
It miss `"php-http/client-implementation": "^1.0"` so the error of not found client will be thrown by composer and not discovery

**Comment by welcoMattic** [comment]
@joelwurtz @Nyholm done

**Comment by joelwurtz** [comment]
LGTM

**Comment by Nyholm** [comment]
Thank you !

**Pull request merged**